### PR TITLE
manual: signature option for caml_example

### PR DESCRIPTION
**Pull request by Octachron**
This small PR extends the options of the `caml_example` pseudo-environment with a signature mode. With this PR, the possible modes for caml_example are:
* `toplevel`, for examples that should look as if they were send to the toplevel
* `verbatim`, for generic code examples
* `signature`, for examples of interface files or signature

This seems to cover most documentation needs in the manual, and it is sufficient to convert the faulty example fixed by @lpw25 in #1693 .
In term of implementation, the code inside the environment is merely wraped inside `module type Wrap = sig ... end` before being send to the ocaml subprocess.

### CODE REVIEW

**Comment by gasche** [comment]
I am not sure whether the `sig .. end` wrapping mechanism works correctly (or whether it is going behave in weird/surprising ways in some cases; what if we use the mode that also shows the toplevel output?). I assume you have tested it by converting one of the signatures in the manual, and checked that the output is as you expect? If so, maybe this change could be part of the patch as well?

**Comment by Octachron** [comment]
For more concrete example, I finished the conversion of the example in ocamldoc: [html](http://www.polychoron.fr/ocaml-nonmanual/pr1702/ocamldoc.html#353) and [pdf](http://www.polychoron.fr/ocaml-nonmanual/pr1702/pdfmanual.pdf) . (The corresponding commit also fixes transf.mll to make sure that \camlexample is considered a verbatim mode).

> what if we use the mode that also shows the toplevel output?

In this case, it would show the wrapping module type, I can see two options to avoid this case: either require to combine the signature mode with `caml_example *` or remove the wrapping module type. I tend for the first option.

**Comment by gasche** [comment]
Yes, the first option is simplest, and inferred signatures are not very useful anyway (they mostly repeat the signature).

Would you consider adding a Changes entry, though? This is a nice change that did require some work from you, and I like the idea of crediting changes to the manual and broader documentation ecosystem.

File: manual/manual/cmds/Makefile
Comment (gasche) [inline manual/manual/cmds/Makefile:54]: We are accumulating logic to use CAMLLATEX/TRANSF/TEXQUOTE in several different Makefiles (apparently `cmds/Makefile` does not use CAMLLATEX but only the other two?). Maybe it would make sense to factorize this in a shared makefile, or as a separate shellscript to invoke?

**Review by gasche — APPROVED** [review]
Approved -- although I understand that you might want to make further tweaks. Feel free to merge whenever you are ready and the CI passes.

(This change caught other mistakes in the `ocamldoc` manual than the ones caught by Leo, which means that it was indeed a reasonable idea.)

File: manual/manual/cmds/Makefile
Comment (Octachron) [reply manual/manual/cmds/Makefile:54]: This is probably a good idea for a next PR.

**Comment by Octachron** [comment]
I added an error message for the combination of `\begin{caml_example}{signature}`, and a change entry. I will merge after the merge of Leo's doctring fix.

**Comment by gasche** [comment]
I think we could keep this in 4.07, as it could be useful for further improve-the-manual changes (eg. a more complete version of #1209) that it could be worthwhile to include in 4.07 documentation. The present PR does not touch anything that is critical for the release preparation/testing, and it is customary to give external tools (ocamldoc, the manual build tools) more slack in freeze decisions.

**Comment by Octachron** [comment]
It is true that this is mostly a validation tool. I moved back the change entry to 4.07 .

File: manual/manual/cmds/Makefile
Comment (gasche) [inline manual/manual/cmds/Makefile]: Does this not need an update to take 482e8a8b52ff9fb8e8f8aecb17264f314b9a37f0 into account?

Comment (Octachron) [reply manual/manual/cmds/Makefile]: You are totally right, I had forgotten this point. I should try to make my test environment sensitive to this problem. Fixed.

**Review by gasche — APPROVED** [review]
Good to merge when CI passes.

**Comment by Octachron** [comment]
I removed a wandering unused  macros, since keeeping the macro count as low as possible seemed important enough to warrant a last minute fix. I will squash and merge once the CI passes.

**Comment by Octachron** [comment]
Cherry-picked to 4.07 as d040ad69826389e21f471fc7

**Pull request merged**
